[Bug]: Comfy UI won't load...

### Frontend Version

![2024-10-09 16_44_59-ComfyUI - Brave](https://github.com/user-attachments/assets/a4169a95-a7a5-462f-b728-edfb3152f6f7)

when i run the bat it's just shows the spinning wheel forever.

here's a screenshot of the issue... ![2024-10-09 17_57_59-](https://github.com/user-attachments/assets/3698255a-ac22-4ba2-aeeb-b4e711b9447e)



### Expected Behavior

usually it loads the last workflow that was previously loaded...

### Actual Behavior

it just shows this page with a spinning wheel but doesn't load anything.

![2024-10-09 16_44_59-ComfyUI - Brave](https://github.com/user-attachments/assets/a4ed02d4-4663-4dd5-8aee-375239d47b64)


### Steps to Reproduce

i don't really know if it's a node thing, but when i change the front end version line in the bat, it works again...

--front-end-version Comfy-Org/ComfyUI_frontend@1.3.13



### Debug Logs

```powershell
To see the GUI go to: http://127.0.0.1:8188
FETCH DATA from: H:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager\extension-node-map.json [DONE]
H:\AI\ComfyUI\ComfyUI\custom_nodes\comfyui-mixlab-nodes\webApp\lib/photoswipe-lightbox.esm.min.js
H:\AI\ComfyUI\ComfyUI\custom_nodes\comfyui-mixlab-nodes\webApp\lib/photoswipe.min.css
['anime_lora.safetensors', 'art_lora.safetensors', 'disney_lora.safetensors', 'realism_lora.safetensors', 'scenery_lora.safetensors']
['anime_lora.safetensors', 'art_lora.safetensors', 'disney_lora.safetensors', 'realism_lora.safetensors', 'scenery_lora.safetensors']
H:\AI\ComfyUI\ComfyUI\custom_nodes\comfyui-mixlab-nodes\webApp\lib/classic.min.css
H:\AI\ComfyUI\ComfyUI\custom_nodes\comfyui-mixlab-nodes\webApp\lib/juxtapose.css
H:\AI\ComfyUI\ComfyUI\custom_nodes\comfyui-mixlab-nodes\webApp\lib/model-viewer.min.js
H:\AI\ComfyUI\ComfyUI\custom_nodes\comfyui-mixlab-nodes\webApp\lib/pickr.min.js
H:\AI\ComfyUI\ComfyUI\custom_nodes\comfyui-mixlab-nodes\webApp\lib/juxtapose.min.js
['anime_lora.safetensors', 'art_lora.safetensors', 'disney_lora.safetensors', 'realism_lora.safetensors', 'scenery_lora.safetensors']
['anime_lora.safetensors', 'art_lora.safetensors', 'disney_lora.safetensors', 'realism_lora.safetensors', 'scenery_lora.safetensors']
H:\AI\ComfyUI\ComfyUI\custom_nodes\comfyui-mixlab-nodes\webApp\lib/photoswipe.min.css
H:\AI\ComfyUI\ComfyUI\custom_nodes\comfyui-mixlab-nodes\webApp\lib/classic.min.css
H:\AI\ComfyUI\ComfyUI\custom_nodes\comfyui-mixlab-nodes\webApp\lib/juxtapose.css
H:\AI\ComfyUI\ComfyUI\custom_nodes\comfyui-mixlab-nodes\webApp\lib/pickr.min.js.map
H:\AI\ComfyUI\ComfyUI\custom_nodes\comfyui-mixlab-nodes\webApp\lib/model-viewer.min.js.map

this is where it stops...
```


### Browser Logs

api.ts:48 Running on 127.0.0.1:8188
logging.ts:332 ComfyUI Front-end version: 1.3.14
main.js:85 i18next: languageChanged en
main.js:85 i18next: initialized Object
main.js:85 0.1.23  2023-11-30 01:15:32
main.js:102 @mediapipe/pose Module
main.js:102 MyPose ƒ U0(f){var x=this;f=f||{},this.h=new Ff({locateFile:f.locateFile,files:function(M){return[{url:"pose_solution_packed_assets_loader.js"},{simd:!1,url:"pose_solution_wasm_bin.js"},{simd:!0,url:"pose_sol…
app.ts:1722 Error loading extension /extensions/ComfyUI_3dPoseEditor/js/main.js Error: Minified React error #299; visit https://reactjs.org/docs/error-decoder.html?invariant=299 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at gn.createRoot (main.js:40:55756)
    at LV (main.js:3441:30649)
    at main.js:3441:30767
(anonymous) @ app.ts:1722
checkVersion_mixlab.js:12 Latest release version: v0.43.0
api.ts:244 Unhandled message: {"type": "crystools.monitor", "data": {"cpu_utilization": 20.5, "ram_total": 34254286848, "ram_used": 18199080960, "ram_used_percent": 53.1, "hdd_total": 8001201111040, "hdd_used": 5745588568064, "hdd_used_percent": 71.8, "device_type": "cuda", "gpus": [{"gpu_utilization": 1, "gpu_temperature": 51, "vram_total": 12884901888, "vram_used": 1194471424, "vram_used_percent": 9.270318349202473}]}} Error: Unknown message type crystools.monitor
    at WebSocket.<anonymous> (api.ts:239:23)
(anonymous) @ api.ts:244
model-viewer.min.js:17 WARNING: Multiple instances of Three.js being imported.
(anonymous) @ model-viewer.min.js:17
3d_mixlab.js:295 nodeType.comfyClass 3DImage
openposeeditor.js:251 [3D Pose Editor] Registering node... Object
gptSampler.js:9 GPTSampler detected
beforeRegisterNodeDef @ gptSampler.js:9
runtime-core.esm-bundler.js:270 TypeError: output.map is not a function
    at ComfyNodeDefImpl.transformOutputSpec (nodeDefStore.ts:179:27)
    at new ComfyNodeDefImpl (nodeDefStore.ts:173:36)
    at Proxy.updateNodeDefs (nodeDefStore.ts:305:29)
    at Proxy.wrappedAction (pinia.mjs:1405:26)
    at ComfyApp.updateVueAppNodeDefs (app.ts:1962:18)
    at ComfyApp.registerNodes (app.ts:1977:12)
    at async ComfyApp.setup (app.ts:1870:5)
    at async GraphCanvas.vue:136:9
logError @ runtime-core.esm-bundler.js:270
Show 1 more frame
Show less
reroute.js:75 Uncaught (in promise) undefined

### What browsers do you use to access the UI ?

Brave

### Other

_No response_

--- a/src/stores/nodeDefStore.ts
+++ b/src/stores/nodeDefStore.ts
@@ -169,21 +169,21 @@ export class ComfyNodeDefImpl {
     this.deprecated = obj.deprecated ?? obj.category === ''
     this.experimental =
       obj.experimental ?? obj.category.startsWith('_for_testing')
-    this.input = new ComfyInputsSpec(obj.input)
+    this.input = new ComfyInputsSpec(obj.input ?? {})
     this.output = ComfyNodeDefImpl.transformOutputSpec(obj)
     this.nodeSource = getNodeSource(obj.python_module)
   }
 
   private static transformOutputSpec(obj: any): ComfyOutputsSpec {
     const { output, output_is_list, output_name, output_tooltips } = obj
-    const result = output.map((type: string | any[], index: number) => {
+    const result = (output ?? []).map((type: string | any[], index: number) => {
       const typeString = Array.isArray(type) ? 'COMBO' : type
 
       return new ComfyOutputSpec(
         index,
-        output_name[index],
+        output_name?.[index],
         typeString,
-        output_is_list[index],
+        output_is_list?.[index],
         Array.isArray(type) ? type : undefined,
         output_tooltips?.[index]
       )

--- a/src/types/apiTypes.ts
+++ b/src/types/apiTypes.ts
@@ -348,8 +348,8 @@ const zComfyOutputTypesSpec = z.array(
 const zComfyNodeDef = z.object({
   input: zComfyInputsSpec.optional(),
   output: zComfyOutputTypesSpec.optional(),
-  output_is_list: z.array(z.boolean()),
-  output_name: z.array(z.string()),
+  output_is_list: z.array(z.boolean()).optional(),
+  output_name: z.array(z.string()).optional(),
   output_tooltips: z.array(z.string()).optional(),
   name: z.string(),
   display_name: z.string(),

--- a/src/types/apiTypes.ts
+++ b/src/types/apiTypes.ts
@@ -346,8 +346,8 @@ const zComfyOutputTypesSpec = z.array(
 )
 
 const zComfyNodeDef = z.object({
-  input: zComfyInputsSpec,
-  output: zComfyOutputTypesSpec,
+  input: zComfyInputsSpec.optional(),
+  output: zComfyOutputTypesSpec.optional(),
   output_is_list: z.array(z.boolean()),
   output_name: z.array(z.string()),
   output_tooltips: z.array(z.string()).optional(),

--- a/tests-ui/tests/nodeDef.test.ts
+++ b/tests-ui/tests/nodeDef.test.ts
@@ -356,6 +356,20 @@ describe('ComfyNodeDefImpl', () => {
     expect(result.output.all).toEqual([])
   })
 
+  it('should handle undefined fields', () => {
+    const plainObject = {
+      name: 'EmptyOutputNode',
+      display_name: 'Empty Output Node',
+      category: 'Test',
+      python_module: 'test_module',
+      description: 'A node with no outputs'
+    }
+
+    const result = new ComfyNodeDefImpl(plainObject)
+    expect(result.output.all).toEqual([])
+    expect(result.input.all).toEqual([])
+  })
+
   it('should handle complex input specifications', () => {
     const plainObject = {
       name: 'ComplexInputNode',

